src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -629,17 +630,27 @@ public void checkSensitivities(Network network, LfNetwork lfNetwork, List<Sensit
         }
     }
 
-    public void checkContingencies(LfNetwork lfNetwork, List<Contingency> contingencies) {
-        for (Contingency contingency : contingencies) {
-            for (ContingencyElement contingencyElement : contingency.getElements()) {
+    public void checkContingencies(LfNetwork lfNetwork, List<ContingencyContext> contingencies) {
+        for (ContingencyContext contingency : contingencies) {
+            for (ContingencyElement contingencyElement : contingency.getContingency().getElements()) {
                 if (!contingencyElement.getType().equals(ContingencyElementType.BRANCH)) {
                     throw new UnsupportedOperationException("Only contingencies on a branch are yet supported");
                 }
-                LfBranch lfBranch = lfNetwork.getBranchById(contingencyElement.getId());
+            }
+            Set<String> branchesToRemove = new HashSet<>(); // contains the branches that are connected only on one side
```

Comment:
Maybe it could just be an option if the `ContingencyContext` creation to avoid remove this branches afterward?